plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/reporter/CustomReporter.kt

```diff
@@ -0,0 +1,17 @@
+package org.jlleitschuh.gradle.ktlint.reporter
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Dependency
+
+data class CustomReporter(
+    val reporterId: String,
+    val reporterFileExtension: String,
+    val reporterDependencyNotation: Any
+) {
+    internal lateinit var dependency: Dependency
+
+    internal fun resolveDependency(project: Project): Dependency {
+        dependency = project.dependencies.create(reporterDependencyNotation)
+        return dependency
+    }
```

Comment:
Any better way to do this without `lateinit`?
Like what happens if one `CustomReporter` is shared between two projects? I don't know if that's even possible, but it if is, you could end up with a `Dependency` resolved against the wrong project perhaps?

Reply:
Actually it is not needed, removed.